src/System.Private.CoreLib/shared/System/Runtime/CompilerServices/RuntimeHelpers.cs

```diff
@@ -80,5 +80,11 @@ public static void PrepareConstrainedRegions()
         public static void PrepareConstrainedRegionsNoOP()
         {
         }
+
+        internal static unsafe bool IsNullRef<T>(in T value)
```

Comment:
I think this should be on Unsafe: `Unsafe.IsNullRef`. It is in the same category as `Unsafe.SizeOf` - not strictly unsafe, but unsafe from C# point of view. If it proves useful, I think we should add it to the public version of S.R.CS.Unsafe as well.